control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1213,6 +1218,65 @@ func (r *reconciler) reconcileAuthOIDC(ctx context.Context, hcp *hyperv1.HostedC
 	return errors.NewAggregate(errs)
 }
 
+func (r *reconciler) reconcileAuthConfig(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	var errs []error
+	providerReadyStatus := metav1.ConditionTrue
+	providerReadyReason := hyperv1.AsExpectedReason
+	providerReadyMessage := "OIDC providers reconciled."
+
+	authenticationConfig := globalconfig.AuthenticationConfiguration()
+	_, err := r.CreateOrUpdate(ctx, r.client, authenticationConfig, func() error {
+		return globalconfig.ReconcileAuthenticationConfiguration(authenticationConfig, hcp.Spec.Configuration, hcp.Spec.IssuerURL)
+	})
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile authentication config: %w", err))
```

Comment:
I think we should return early here, and not check the secrets on reconciliation error.